modules/reactor/test/tests/meshgenerators/reporting_id/hexagonal_id/assembly_reporting_id_pattern.i

```diff
@@ -0,0 +1,65 @@
+[Mesh]
+  [pin1]
+    type = PolygonConcentricCircleMeshGenerator
+    preserve_volumes = true
+    ring_radii = 0.4
+    ring_intervals = 2
+    background_intervals = 1
+    num_sides = 6
+    num_sectors_per_side = '2 2 2 2 2 2'
+    polygon_size = 0.5
+  []
+  [pin2]
+    type = PolygonConcentricCircleMeshGenerator
+    preserve_volumes = true
+    ring_radii = 0.4
+    ring_intervals = 2
+    background_intervals = 1
+    num_sides = 6
+    num_sectors_per_side = '2 2 2 2 2 2'
+    polygon_size = 0.5
+  []
+  [assembly]
+    type = HexIDPatternedMeshGenerator
+    inputs = 'pin1 pin2'
+    pattern_boundary = hexagon
+    pattern = '  1 0 1;
+                0 0 0 0;
+               1 0 1 0 1;
+                0 0 0 0;
+                 1 0 1'
+    hexagon_size = 2.6
+    duct_sizes = '2.4 2.5'
+    duct_intervals = '1 1'
+    id_name = 'pin_id'
+    assign_type = 'pattern'
```

Comment:
these 3 input files (maybe only 2) can be merged.
then in the tests spec, pass a command line argument:
cli_args = 'Mesh/assembly/assign_type=pattern'